### PR TITLE
[Merged by Bors] - Prepare crevice for vendored release

### PR DESCRIPTION
**Pull request by mockersf**
# Objective

- Our crevice is still called "crevice", which we can't use for a release
- Users would need to use our "crevice" directly to be able to use the derive macro

## Solution

- Rename crevice to bevy_crevice, and crevice-derive to bevy-crevice-derive
- Re-export it from bevy_render, and use it from bevy_render everywhere
- Fix derive macro to work either from bevy_render, from bevy_crevice, or from bevy

## Remaining

- It is currently re-exported as `bevy::render::bevy_crevice`, is it the path we want?
- After a brief suggestion to Cart, I changed the version to follow Bevy version instead of crevice, do we want that?
- Crevice README.md need to be updated
- in the `Cargo.toml`, there are a few things to change. How do we want to change them? How do we keep attributions to original Crevice?
```
authors = ["Lucien Greathouse <me@lpghatguy.com>"]
documentation = "https://docs.rs/crevice"
homepage = "https://github.com/LPGhatguy/crevice"
repository = "https://github.com/LPGhatguy/crevice"
```

### CODE REVIEW

File: crates/bevy_crevice/README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ uniform MAIN {
 ```
 
 ```rust
-use crevice::std140::{AsStd140, Std140};
```
Comment (bjorn3) [inline crates/bevy_crevice/README.md:53]: Can you add a note to the readme that this is a fork of the crevice maintained by bevy.

Comment (mockersf) [reply crates/bevy_crevice/README.md:53]: Yup it's in the "remaining", but I would like Cart to chime in on how he wants to present that, or at least someone with better English

**Comment by cart** [comment]
Great work! In general this looks good to me. I think the readme should cover the following topics (in this order):

* This is a Bevy fork of `crevice`
* Non bevy users should probably consume upstream `crevice` (with a link to it)
* Explain rationale for the fork / the changes we have made:
  * Easier derive macro usage for bevy users
  * We are currently using unmerged features (ex: the [crevice arrays pr](https://github.com/LPGhatguy/crevice/pull/27/))
  * Might ultimately rename types to meet Bevy needs (ex: `derive(Uniform)` instead of `derive(AsStd140)`)

I think we should leave attributions as they are for now, with the exception of the repo link, which should point to the bevy repo.

File: crates/bevy_pbr/src/render/mod.rs [hunk 2]
```diff
@@ -13,6 +13,7 @@ use bevy_ecs::{
     system::{lifetimeless::*, SystemParamItem},
 };
 use bevy_render::{
+    bevy_crevice::std140::AsStd140,
```
Comment (cart) [inline crates/bevy_pbr/src/render/mod.rs:16]: I'm thinking we probably want to _just_  re-export AsStd140, AsStd430, Std140, and Std430 under `render_resource` to make importing more straightforward.

Comment (mockersf) [reply crates/bevy_pbr/src/render/mod.rs]: this makes the macro to get to the correct path in the derive a little more complicated, but done 👍

**Comment by mockersf** [comment]
I updated the readme 👍

**Comment by mockersf** [comment]
Ooh two compiler panics in CI, fancy!

**Comment by mockersf** [comment]
rust nightly ICE probably related to one of https://github.com/rust-lang/rust/issues/92187 https://github.com/rust-lang/rust/issues/92163 and could be fixed by https://github.com/rust-lang/rust/pull/92175

**Review by MinerSebas — CHANGES_REQUESTED** [review]
This crate does also need to be added to the `/tools/publish.sh`

**Pull request closed without merging**